### ISSUE TITLE
Script ends prematurely

### ISSUE THREAD

**Issue by horca** [hertg/egpu-switcher#34]
My setup: Arch Linux, Razer Blade Stealth 2019 Mercury White (without any DGPU), Razer Core X, RX 5700 XT.

Problem: Whenever I run the egpu-switcher script, it somehow ends prematurely inside this block:
```bash
for disp in ${disp_path}; do
	if [ -e $disp ]; then
		disp_num=$(expr $disp_num + 1)
		disp_disconnect=$(expr $disp_disconnect + $(cat $disp | grep -ce ^disconnected$))
	fi
done
```
Bash debug output ends with this:
```
+ echo -e '\033[1;34m[info]\033[0m EGPU is \033[1;32mconnected\033[0m.'
[info] EGPU is connected.
+ gpu_connected=1
+ '[' 1 = 1 ']'
+ mode=egpu
+ '[' egpu = egpu ']'
++ grep -ce nvidia /etc/X11/xorg.conf.egpu
+ '[' 0 -eq 0 ']'
++ cat /etc/X11/xorg.conf.egpu
++ grep -Ei BusID
++ grep -oEi '[0-9]+\:[0-9]+\:[0-9]+'
+ declare egpu_pci_id=48:0:0
+ busArray=(${egpu_pci_id//:/ })
+ declare busArray
+ declare bus1d=48
+ declare bus2d=0
+ declare bus3d=0
++ printf %02x 48
+ declare bus1h=30
++ printf %02x 0
+ declare bus2h=00
++ printf %01x 0
+ declare bus3h=0
++ echo 30:00.0
+ declare hex_id=30:00.0
+ declare 'disp_path=/sys/bus/pci/devices/[0-9a-f:]*30:00.0/drm/card[0-9]*/card[0-9]*-*/status'
+ declare disp_num=0
+ declare disp_disconnect=0
+ for disp in ${disp_path}
+ '[' -e /sys/bus/pci/devices/0000:30:00.0/drm/card1/card1-DP-6/status ']'
++ expr 0 + 1
+ disp_num=1
+++ cat /sys/bus/pci/devices/0000:30:00.0/drm/card1/card1-DP-6/status
+++ grep -ce '^disconnected$'
++ expr 0 + 0
+ disp_disconnect=0
```
and then it ends.

Bash is really not my cup of tea so maybe I'm missing something? Workaround for me is to just delete the whole disconnected display verification block.

**Comment by hertg** [comment]
@ewagner12 
I believe the error occurs in the code you commited via PR #14 and #17.
Also, i refactored some of your code in a later commit, maybe my changes broke something.
I can't really test the functionality you've added, because I only have NVIDIA GPUs.
Would be great if you could look into that issue :)

Relevant source code:
https://github.com/hertg/egpu-switcher/blob/master/egpu-switcher#L355-L389

**Comment by ewagner12** [comment]
I'll test this more and see if I can replicate this

**Comment by ewagner12** [comment]
@horca 
Can I see the output of the commands
`cat /sys/bus/pci/devices/[0-9a-f:]*30:00.0/drm/card[0-9]*/card[0-9]*-*/status`
and
`ls /sys/bus/pci/devices/[0-9a-f:]*30:00.0/drm/card[0-9]*/`
on my card this shows all 3 physical outputs and their status, it looks like the script is only parsing one of yours, and I'm guessing your 5700 XT has more than one physical output

**Comment by horca** [comment]
`cat /sys/bus/pci/devices/[0-9a-f:]*30:00.0/drm/card[0-9]*/card[0-9]*-*/status`
```
connected
connected
disconnected
disconnected
```

` ls /sys/bus/pci/devices/[0-9a-f:]*30:00.0/drm/card[0-9]*/`
```
card1-DP-6  card1-DP-8      dev     power      uevent
card1-DP-7  card1-HDMI-A-2  device  subsystem

```

**Comment by ewagner12** [comment]
Ok, I think I was able to replicate and solve this, let me know if this works for you, @horca. Replace the original problematic for loop you mentioned with:

```
for disp in ${disp_path}; do
	if [ -e $disp ]; then
		((disp_num = $disp_num + 1)) || true
		((disp_disconnect = $disp_disconnect + $(cat $disp | grep -ce ^disconnected$))) || true
	fi
done
```

**Comment by horca** [comment]
@ewagner12 Tried it and its working without any problem.

**Comment by hertg** [comment]
I've published a new release 0.14.0 which includes this fix.
Thanks for your work @ewagner12!
